oresat_dxwifi/resources/temperature.py

```diff
@@ -34,29 +32,22 @@ def __init__(self, adc_thermistor_pin: int = 6, is_mock_adc: bool = False):
             is_mock_adc: False = real world ADC, True = mock ADC.
         """
         super().__init__()
-        self.index = self.TEMPERATURE_INDEX
+
         self.adc = Adc(adc_thermistor_pin, is_mock_adc)
 
     def on_start(self):
         """Sets up an SDO read callback"""
-        self.node.add_sdo_read_callback(self.index, self.on_read)
 
-    def on_read(self, index: int, subindex: int) -> float:
-        """Returns temperature value
+        self.node.add_sdo_callbacks("radio", "temperature", self._on_read_temperature, None)
 
-        Args:
-            index: Index in Object Dictionary
-            subindex: Subindex of "index" argument
+    def _on_read_temperature(self) -> int:
+        """Returns temperature value
 
         Returns:
             ret: Calculated temperature value
         """
-        ret = None
-
-        if index == self.index:
-            ret = self.find_temperature()
 
-        return ret
+        return int(self.find_temperature())
```

Comment:
Why are we casting to int now? Does OLAF expect an int?

Reply:
It was defined as a int8 the OD, not a float.

Reply:
That doesn't seem right. It should have been a real32. I tried eds-editor on oresat_dxwifi.dcf and oresat_dxiwifi.eds and it shows up as such. Is it supposed to be an int8 in its use case?

Reply:
Yea, I doubt the sensor has the accuracy for a float (I could be wrong). We usually use int8 for temperatures values as the decimal part of the float value is not needed.